core/src/consensus/tendermint/worker.rs

```diff
@@ -454,7 +454,7 @@ impl Worker {
         self.validators.next_block_proposer(prev_block_hash, view)
     }
 
-    pub fn proposal_at(&self, height: Height, view: View) -> Option<(SchnorrSignature, usize, Bytes)> {
+    pub fn first_proposal_at(&self, height: Height, view: View) -> Option<(SchnorrSignature, usize, Bytes)> {
```

Comment:
`VoteCollector` and `StepCollector` don't guarantee the ordering of underlying votes. You should change the prefix to `any_` or fix the vote collector

Reply:
I made the vote_collecter returns messages in the insertion order.